client/src/sections/material/MaterialForm.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { Fragment, useEffect } from 'react';
+import { type SubmitHandler, useForm } from 'react-hook-form';
+import Button from '@/src/shared/components/Button';
+import DivCollapse from '@/src/shared/components/Collapse/DivCollapse';
+import RFInputField from '@/src/shared/components/ReactForm/RFInputField';
+import RFSelectField from '@/src/shared/components/ReactForm/RFSelectField';
+import RFTextField from '@/src/shared/components/ReactForm/RFTextField';
+import type { MaterialFormInput } from '@/src/shared/utils';
+
+interface TestingProps {
+  activeField: string;
+}
+
+const MaterialForm: React.FunctionComponent<TestingProps> = ({
+  activeField
+}: TestingProps) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<MaterialFormInput>();
+
+  const onSubmit: SubmitHandler<MaterialFormInput> = async (
+    data: MaterialFormInput
+  ): Promise<void> => {
+    reset();
+    console.log(data);
+  };
+
+  const folders = [
+    { value: '0', label: 'Nothing Selected' },
+    { value: '1', label: 'Example Folder 1' },
+    { value: '2', label: 'Example Folder 2' },
+    { value: '3', label: 'Example Folder 3' }
+  ];
+
+  const categories = [
+    { value: '0', label: 'Nothing Selected' },
+    { value: '1', label: 'Example Category 1' },
+    { value: '2', label: 'Example Category 2' },
+    { value: '3', label: 'Example Category 3' }
+  ];
+
+  const courses = [
+    { value: '0', label: 'Nothing Selected' },
+    { value: '1', label: 'Example Course 1' },
+    { value: '2', label: 'Example Course 2' },
+    { value: '3', label: 'Example Course 3' }
+  ];
+
+  const selectFields = [
+    {
+      label: 'Assign to Folder',
+      id: 'directory',
+      options: folders
+    },
+    {
+      label: 'Assign to Category',
+      id: 'category',
+      options: categories
+    },
+    {
+      label: 'Assign to Course',
+      id: 'course',
+      options: courses
+    }
+  ];
+
+  useEffect(() => {
+    reset();
+  }, [activeField, reset]);
+
+  return (
+    <Fragment>
+      <div className="p-4 max-h-[800px] overflow-y-auto">
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div className="mb-4">
+            {activeField === 'singleFile' && (
+              <div>
+                <h2 className="text-md text-blue-900 mb-5">Upload a File</h2>
+
+                <RFInputField
+                  className="border-none"
+                  type="file"
+                  {...register('link', { required: true })}
+                  error={!!errors.link && 'This field is required'}
+                />
```

Comment:
```suggestion
                <RFInputField
                  label="Upload a File"
                  className="border-none"
                  type="file"
                  {...register('link', { required: true })}
                  error={!!errors.link && 'This field is required'}
                />
```

Please use label prop for the title, if you want to add different custom style to it, please add `labelClass` prop to  `RFInputField` component so you can add custom style to label